src/Link.tsx

```diff
@@ -38,5 +41,41 @@ const Link = styled.a<StyledLinkProps>`
   ${sx};
 `
 
+const Link = forwardRef(({as: Component = 'a', ...props}, forwardedRef) => {
+  const innerRef = React.useRef<HTMLAnchorElement>(null)
+  useRefObjectAsForwardedRef(forwardedRef, innerRef)
+
+  if (__DEV__) {
+    /**
+     * The Linter yells because it thinks this conditionally calls an effect,
+     * but since this is a compile-time flag and not a runtime conditional
+     * this is safe, and ensures the entire effect is kept out of prod builds
+     * shaving precious bytes from the output, and avoiding mounting a noop effect
+     */
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      if (
+        innerRef.current &&
+        !(innerRef.current instanceof HTMLButtonElement) &&
+        !(innerRef.current instanceof HTMLAnchorElement)
+      ) {
+        // eslint-disable-next-line no-console
+        console.warn('This component should be an instanceof a semantic button or anchor')
```

Comment:
Can we be more specific about which component is causing this warning? If I just saw `This component...` in the console, it would be hard to track down. If we log the ref too, it might be easier to find the corresponding component.

```suggestion
        console.error('Error: Found `Link` component that renders an inaccessible element', innerRef.current, 'Please ensure `Link` always renders as <a> or <button>')
```